elastic/shared/runners/dlm.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import asyncio
+import logging
+
+from shared.utils.track import mandatory
+
+"""
+Provides operations for Data Lifecycle Management (DLM) on data streams.
+"""
+
+
+async def put_lifecycle(es, params):
+    """
+    Configure data lifecycle management on a data stream.
+
+    Parameters:
+    - data-stream: Name of the data stream (supports wildcards)
+    - data_retention: Optional retention period (e.g., "7d", "30d")
+    - enabled: Optional boolean to enable/disable lifecycle (default: true)
+    - downsampling: Optional dict for downsampling configuration
+
+    Returns: Number of data streams configured
+    """
+    logger = logging.getLogger(__name__)
+    data_stream = mandatory(params, "data-stream", "put-lifecycle")
+
+    # Build lifecycle configuration
+    lifecycle_config = {}
+
+    if "data_retention" in params:
+        lifecycle_config["data_retention"] = params["data_retention"]
+
+    if "enabled" in params:
+        lifecycle_config["enabled"] = params["enabled"]
+
+    if "downsampling" in params:
+        lifecycle_config["downsampling"] = params["downsampling"]
+
+    logger.info(f"Configuring lifecycle on data stream [{data_stream}]: {lifecycle_config}")
+
+    # Get all data streams matching the pattern
```

Comment:
If we do add this API, we don't need to resolve the data streams ourselves as the API already supports comma-separated data stream names (including wildcards): https://github.com/elastic/elasticsearch/blob/4e91c46bbf8333e76051919b455dca25ba9ec408/server/src/main/java/org/elasticsearch/action/datastreams/lifecycle/PutDataStreamLifecycleAction.java#L40-L40